src/components/button/button.ts

```diff
@@ -0,0 +1,23 @@
+import Block from 'core/Block';
+
+import './button.scss';
+
+interface ButtonProps {
+  text: string;
+  type: string;
```

Comment:
Для таких данных, как тип чего-то, ```string``` не очень подходит. Тип - это ограниченное количество значений. ```string``` допускает любые значения. Тут лучше подойдет использование union'a
```suggestion
  type?: "button" | "submit";
```